diff --git a/node_modules/multiformats/types/src/cid.d.ts b/node_modules/multiformats/types/src/cid.d.ts
index 5b1e009..1a26729 100644
--- a/node_modules/multiformats/types/src/cid.d.ts
+++ b/node_modules/multiformats/types/src/cid.d.ts
@@ -157,6 +157,6 @@ export class CID {
 }
 export type MultihashDigest = import('./hashes/interface').MultihashDigest;
 export type CIDVersion = 0 | 1;
-export type MultibaseEncoder<Prefix> = import('./bases/interface').MultibaseEncoder<Prefix>;
-export type MultibaseDecoder<Prefix> = import('./bases/interface').MultibaseDecoder<Prefix>;
+export type MultibaseEncoder<Prefix extends string> = import('./bases/interface').MultibaseEncoder<Prefix>;
+export type MultibaseDecoder<Prefix extends string> = import('./bases/interface').MultibaseDecoder<Prefix>;
 //# sourceMappingURL=cid.d.ts.map
\ No newline at end of file
